core/src/main/java/tc/oc/pgm/util/Players.java

```diff
@@ -34,10 +34,16 @@ public static String getVisibleName(CommandSender viewer, Player other) {
   }
 
   public static List<String> getPlayerNames(CommandSender sender, String query) {
-    return Bukkit.getOnlinePlayers().stream()
-        .filter(p -> Players.isVisible(sender, p))
-        .map(p -> Players.getVisibleName(sender, p))
-        .filter(n -> LiquidMetal.match(n, query))
+    List<String> playerSuggestions =
+        Bukkit.getOnlinePlayers().stream()
+            .filter(p -> Players.isVisible(sender, p))
+            .map(p -> Players.getVisibleName(sender, p))
+            .filter(n -> LiquidMetal.match(n, query))
+            .collect(Collectors.toList());
+    boolean prefixMatchFound =
+        playerSuggestions.stream().anyMatch((suggestion) -> suggestion.startsWith(query));
```

Comment:
Chech starts with, but ignoring case
Also, instead of streaming for a boolean, just stream to a list, if its empty return the prev one, if non empty return the new one

Reply:
Right that would be more efficient